tests/compilers/config.rs

```diff
@@ -0,0 +1,146 @@
+use std::sync::Arc;
+use wasmer::{CompilerConfig, Engine as WasmerEngine, Features, ModuleMiddleware, Store};
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Compiler {
+    LLVM,
+    Cranelift,
+    Singlepass,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Engine {
+    Native,
+    JIT,
+}
+
+#[derive(Clone)]
+pub struct Config {
+    pub compiler: Compiler,
+    pub engine: Engine,
+    pub features: Option<Features>,
+    pub middlewares: Vec<Arc<dyn ModuleMiddleware>>,
+    pub canonicalize_nans: bool,
+}
+
+impl Config {
+    pub fn new(engine: Engine, compiler: Compiler) -> Self {
+        Self {
+            compiler,
+            engine,
+            features: None,
+            canonicalize_nans: false,
+            middlewares: vec![],
+        }
+    }
+
+    pub fn set_middlewares(&mut self, middlewares: Vec<Arc<dyn ModuleMiddleware>>) {
+        self.middlewares = middlewares;
+    }
+
+    pub fn set_features(&mut self, features: Features) {
+        self.features = Some(features);
+    }
+
+    pub fn set_nan_canonicalization(&mut self, canonicalize_nans: bool) {
+        self.canonicalize_nans = canonicalize_nans;
+    }
+
+    pub fn store(&self) -> Store {
+        let compiler_config = self.compiler_config(self.canonicalize_nans);
+        let engine = self.engine(compiler_config);
+        Store::new(&*engine)
+    }
+
+    pub fn headless_store(&self) -> Store {
+        let engine = self.engine_headless();
+        Store::new(&*engine)
+    }
+
+    pub fn engine(&self, compiler_config: Box<dyn CompilerConfig>) -> Box<dyn WasmerEngine> {
+        #[cfg(not(feature = "engine"))]
+        compile_error!("Plese enable at least one engine via the features");
+        match &self.engine {
+            #[cfg(feature = "native")]
+            Engine::Native => {
+                let mut engine = wasmer_engine_native::Native::new(compiler_config);
+                if let Some(ref features) = self.features {
+                    engine = engine.features(features.clone())
+                }
+                Box::new(engine.engine())
+            }
+            #[cfg(feature = "jit")]
+            Engine::JIT => {
+                let mut engine = wasmer_engine_jit::JIT::new(compiler_config);
+                if let Some(ref features) = self.features {
+                    engine = engine.features(features.clone())
+                }
+                Box::new(engine.engine())
+            }
+            #[cfg(not(any(feature = "native", feature = "jit")))]
+            engine => panic!(
+                "The {:?} Engine is not enabled. Please enable it using the features",
+                engine
+            ),
+        }
```

Comment:
The `cfg` logic doesn't look entirely correct here, if you try to compile only one engine, I'd expect a compiler error about inexhaustive match. Perhaps change `not(any(engine))` into `not(all(engine))`